lib/note-content-editor.tsx

```diff
@@ -425,6 +427,42 @@ class NoteContentEditor extends Component<Props> {
     this.editor = editor;
     this.monaco = monaco;
 
+    monaco.languages.registerLinkProvider('plaintext', {
+      provideLinks: (model) => {
+        const matches = model.findMatches(
+          'simplenote://note/[a-zA-Z0-9-]+',
+          true, // searchOnlyEditableRange
+          true, // isRegex
+          false, // matchCase
+          null, // wordSeparators
+          false // captureMatches
+        );
+        return {
+          // don't set a URL on these links, because then Monaco skips resolveLink
+          // @cite: https://github.com/Microsoft/vscode/blob/8f89095aa6097f6e0014f2d459ef37820983ae55/src/vs/editor/contrib/links/getLinks.ts#L43:L65
+          links: matches.map(({ range }) => ({ range })),
+        };
+      },
+      resolveLink: (link) => {
+        const href = editor.getModel()?.getValueInRange(link.range);
+        if (!href) {
+          return;
+        }
```

Comment:
here's a stylistic nit but using the newer optional operators I find it nice to collapse multiple checks into one.

```ts
const href = editor.getModel()?.getValueInRange(link.range) ?? '';
const match = /…/.exec(href);
if (!match) {
	return;
}
```

that is, we don't care so much in this function if the `href` is empty so much as we care if it positively matches the internal link format. we can remove the special-casing for the empty link and preserve the same logic as long as we fallback to the empty string.

as I write and test that out though I realize we don't even need this. `/pattern/.exec(undefined)` and `/pattern/.exec(null)` all return `null` so we don't even  need the `?? ''` to make this focus on positively identifying matches over special-casing non-matches

Reply:
I think I only did this because TypeScript was warning me about it

Reply:
I don't think there's any way for this function to be called with an empty match

Reply:
> I don't think there's any way for this function to be called with an empty match

I believe that but we don't carry that information in the type system unfortunately, and hey, it's JavaSript so it could happen. we could assert that it matches but why not handle failures anyway.

`?? ''` should make TypeScript happy as well as clarify that we're matching against a string right?